### DISCUSSION TITLE
Does remix do javascript code splitting at runtime ?

### DISCUSSION THREAD

**Discussion by gerardnico** [remix-run/remix#1450] [Q&A] [answered]
Hallo,

I'm looking into the possibility of migrating a php application into a javascript framework.

As of today, our application:
  * parses a markup document (markdown like) into component 
  * and then adds in the html head the needed resources for each component (javascript, css)

Because we can have a lot of component, we don't want to bundle all javascript in one big file.

I can see in the [compiler documentation](https://remix.run/docs/en/v1/pages/technical-explanation#compiler) that there is some kind of code splitting but I don't understand how it works / what it means.

Does Remix create the code splitting at runtime or do we need to do it at build time ?

Thanks and happy new year
Nico

**Comment by sergiodxa** [comment, marked as answer]
Remix code split your app per route, so each route is a bundle, it also move also modules in your code to their own JS files for better cacheability between routes, this is all automatic.

**Reply by gerardnico** [reply]
Thanks. 

Can we create the routes dynamically ?

For this application:
- a call to the url path `/docs/category/page` gives the location of the markdown/mdx file `/docs/category/page.md`
- a call to the url path `/blog/2002-10-20` gives the location of the markdown/mdx file `/blog/2002-10-20.md`

The steps are:
- the file is parsed, 
- we get a parse tree
- we render it adding the needed resources (js, css) only for the component in the parse tree. 

For instance:
- if we found a comment/discussion component in the mdx file, we add this component javascript
- if we found a user feedback module in the mdx file, we add this component javascript
- if we found a carrousel in the mdx file, we add this component javascript

As of today, what i understand is that we need to use the [remix splat route](https://remix.run/docs/en/v1/api/conventions#splat-routes), there is therefore no notion of route.

**Reply by sergiodxa** [reply]
You need to include params on the route for the dynamic parts

**Reply by gerardnico** [reply]
Thanks. 

I don't really understand how the fact that my page has a dynamic rendering components such as a carrousel, animation should ends up in the route. My understanding is that the rendering of a markup/md/mdx file is seen as one big component.
Only the code layout of the page is splited.

I'm going to make a rapid test. Thanks for the info.
